packages/frontend/src/components/RefreshServer/RefreshStepsModal/index.tsx

```diff
@@ -0,0 +1,165 @@
+    jobStatus: 'RUNNING', // refresh status ('DONE' | 'RUNNING' | 'ERROR')
+    jobResults: '/projects/{projectUuid}/explores',
+    steps: [
+        {
+            name: 'Clone dbt project from github', // step name
+            stepStatus: 'DONE', // step status ('DONE' | 'RUNNING' | 'ERROR' | 'PENDING' | 'SKIPPED')
+            runningTime: '00:00:31', // time it has run for in mins/secs
+        },
+        {
+            name: 'Install dbt project dependencies ', // step name
+            stepStatus: 'RUNNING', // step status ('DONE' | 'RUNNING' | 'ERROR' | 'PENDING' | 'SKIPPED')
+            runningTime: '00:00:31', // time it has run for in mins/secs
+        },
+        {
+            name: 'Compile dbt project', // step name
+            stepStatus: 'ERROR', // step status ('DONE' | 'RUNNING' | 'ERROR' | 'PENDING' | 'SKIPPED')
+            error: {
+                name: 'Name n_stores not found inside',
+                message: 'campaign_all_brand-hist at [2:33]',
+            },
+            runningTime: '00:00:31', // time it has run for in mins/secs
+        },
+        {
+            name: 'Get latest data warehouse schema', // step name
+            stepStatus: 'PENDING', // step status ('DONE' | 'RUNNING' | 'ERROR' | 'PENDING' | 'SKIPPED')
+            runningTime: '00:00:31', // time it has run for in mins/secs
+        },
+        {
+            name: 'Compile metrics and dimensions', // step name
+            stepStatus: 'PENDING', // step status ('DONE' | 'RUNNING' | 'ERROR' | 'PENDING' | 'SKIPPED')
+            runningTime: '00:00:31', // time it has run for in mins/secs
+        },
+    ],
+};
+
+interface Props {
+    closeModal: Dispatch<SetStateAction<boolean>>;
+    isOpen: boolean;
+}
+
+const RefreshStepsModal: FC<Props> = ({ closeModal, isOpen }) => {
+    const statusInfo = (
+        status: string,
+    ): { title: string; icon: IconName; status: string } => {
+        switch (status) {
+            case 'DONE':
+                return {
+                    title: 'Sync successful!',
+                    icon: 'tick-circle',
+                    status: 'Success',
+                };
+            case 'ERROR':
+                return {
+                    title: 'Error in sync',
+                    icon: 'warning-sign',
+                    status: 'Error',
+                };
+            case 'RUNNING':
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'In progress',
+                };
+
+            case 'PENDING':
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'Queued',
+                };
+
+            default:
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'Success',
+                };
+        }
+    };
```

Comment:
We should turn this into a useMemo to avoid calling this function multiple times in the same render loop.